clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1030,6 +1030,7 @@ public void assign(Collection<TopicPartition> partitions) {
      * @return The fetched records (may be empty)
      */
     private Map<TopicPartition, List<ConsumerRecord<K, V>>> pollOnce(long timeout) {
+        client.maybeTriggerWakeup();
```

Comment:
@hachikuji should I adjust the synchronization to the way it's done in the above comment:

```java
    public void maybeTriggerWakeup() {
        synchronized (this.wakeup) {
            if (wakeupDisabledCount == 0 && wakeup.get()) {
                log.trace("Raising wakeup exception in response to user wakeup");
                wakeup.set(false);
                throw new WakeupException();
            }
        }
    }
```

? I didn't want to push over this discussion to avoid it being hidden, but @ijuma is right, sync on `this` potentially comes with deadlocks right?